NuRadioReco/utilities/dataservers.py

```diff
@@ -0,0 +1,115 @@
+    return ranked_dataservers
+
+def download_from_dataserver(remote_path, target_path, unpack_tarball=True, dataservers=dataservers, try_ordered=False):
+    """ download remote_path to target_path from the list of NuRadio dataservers """
+
+    folder = os.path.dirname(target_path)
+    if not os.path.exists(folder):
+        os.makedirs(folder)
+
+    lockfile = target_path+".lock"
+    lock = filelock.FileLock(lockfile)
+   
+    logger.warning(f"Assuring no other process is downloading. Will wait until {lockfile} is unlocked.")
+    with lock:
+        if os.path.isfile(target_path):
+            logger.warning(f"{target_path} already exists. Maybe download was already completed by another instance?")
+            return
+        elif unpack_tarball and (len(glob(os.path.dirname(target_path) + "/*.dat")) > 0): #just check if any .dat files present (similar to NuRadioProposal.py)
+            logger.warning(f"{os.path.dirname(target_path)} contains .dat files. Maybe download was already completed by another instance?")
+            return
+
+        if try_ordered:
+            dataservers = get_available_dataservers_by_responsetime(dataservers)
+            # alternatively:
+            # dataservers = get_available_dataservers_by_timezone(dataservers)
+        requests_status = requests.codes["not_found"]
+        for dataserver in dataservers:
+            URL = f'{dataserver}/{remote_path}'
+
+            logger.warning(
+                "downloading file {} from {}. This can take a while...".format(target_path, URL))
+
+            try:
+                r = requests.get(URL)
+                r.raise_for_status()
+                requests_status = r.status_code
+                break
+            except requests.exceptions.HTTPError as errh:
+                logger.warning(f"HTTP Error for {dataserver}. Does the file {remote_path} exist on the server?")
+                pass
+            except requests.exceptions.ConnectionError as errc:
+                logger.warning(f"Error Connecting to {dataserver}. Maybe you don't have internet... or the server is down?")
+                pass
+            except requests.exceptions.Timeout as errt:
+                logger.warning(f"Timeout Error for {dataserver}.")
+                pass
+            except requests.exceptions.RequestException as err:
+                logger.warning(f"An unusual error for {dataserver} occurred:", err)
+                pass
+
+            logger.warning("problem downloading file {} from {}. Let's see if there is another server.".format(target_path, URL))
+
+        if requests_status != requests.codes["ok"]:
+            logger.error(f"error in download of file {target_path}. Tried all servers in {dataservers} without success.")
+            raise IOError
+
+        with open(target_path, "wb") as code:
+            code.write(r.content)
+        logger.warning("...download finished.")
+
+        if unpack_tarball and target_path.endswith(".tar.gz"):
```

Comment:
FYI unpack archive supports additional archive formats. Not sure if it would make sense to support .tar.xz to save some bandwidth in some cases?

Reply:
Good, but I don't think this is really needed. the proposal tarballs are the only tarballed files, and we chose to have them as tar.gz... they are not so big, and you only need to download once for a NuRadioMC installation.